Sources/SimpleHTTP/Request/MultipartFormData.swift

```diff
@@ -0,0 +1,210 @@
+import Foundation
+import UniformTypeIdentifiers
+
+#if os(iOS) || os(watchOS) || os(tvOS)
+import MobileCoreServices
+#elseif os(macOS)
+import CoreServices
+#endif
+
+struct Header: Hashable {
+
+    let name: HTTPHeader
+    let value: String
+
+}
+
+enum EncodingCharacters {
+    static let crlf = "\r\n"
+}
+
+enum Boundary {
+
+    enum `Type` {
+        case initial
+        case encapsulated
+        case final
+    }
+
+    static func random() -> String {
+        UUID().uuidString
+    }
+
+    static func string(for type: Boundary.`Type`, boundary: String) -> String {
+        switch type {
+        case .initial:
+            return "--\(boundary)\(EncodingCharacters.crlf)"
+        case .encapsulated:
+            return "\(EncodingCharacters.crlf)--\(boundary)\(EncodingCharacters.crlf)"
+        case .final:
+            return "\(EncodingCharacters.crlf)--\(boundary)--\(EncodingCharacters.crlf)"
+        }
+    }
+
+    static func data(for type: Boundary.`Type`, boundary: String) -> Data {
+        let boundaryText = Self.string(for: type, boundary: boundary)
+
+        return Data(boundaryText.utf8)
+    }
+}
+
+class BodyPart {
```

Comment:
anything specific requiring this type to be a class rather than a struct?

Reply:
change in the last commit